developing/vcs/git-repos.md

```diff
@@ -65,47 +65,40 @@ If you need to split out code from one of your repositories into its own
 separate repo, you can follow these steps to make sure you're preserving
 the commit history when doing so.
 
+The `filter-branch` subcommand to `git` can be used to accomplish this, but the
+[documentation](https://git-scm.com/docs/git-filter-branch) strongly
+discourages its use. Instead, a third-party utility called
+[git-filter-repo](https://github.com/newren/git-filter-repo) is recommended.
+
 In your terminal clone a copy of the original repo into a new folder:
 
 ```sh
-git clone git@github.com:OWNER/REPONAME.git NEWREPO
+git clone git@github.com:OWNER/REPONAME.git NEWREPO && cd NEWREPO
 ```
 
-In that new repo folder remove the origin:
+Filter out commits that change the specified directory:
 
 ```sh
-git remote rm origin
+git-filter-repo --subdirectory-filter DIRNAME
 ```
 
-Filter out commits that change the specified directory:
+The `origin` remote will be removed in this process. Add it back with the new
+repo URL, and push the filtered work tree:
 
 ```sh
-git filter-branch --prune-empty --subdirectory-filter DIRNAME main
+git remote add origin git@github.com:OWNER/NEWREPO.git
+git push --set-upstream origin <main>
```

Comment:
Minor nitpick: this refers to the "new repo URL". Move the steps about creating the new repo to before these instructions?

Reply:
I don't know what to make of line 93. I think anyone with the user story "As an engineer, I want to move files from one repository into another while preserving work tree history" knows how to create a repository on github. And if they don't, this line doesn't help. Why not just delete it?

Reply:
Its inclusion in the current version of the doc makes the sequence of steps clear. And retaining it helps to set up the discussion about the repo license and importing the repo into Terraform. Maybe the other PR reviewers have thoughts.